war/src/main/java/io/lumeer/engine/controller/ProjectFacade.java

```diff
@@ -50,68 +66,404 @@ public void setCurrentProjectId(final String projectId) {
    }
 
+   /**
+    * Renames project
+    *
+    * @param projectId
+    *       id of the project to rename
+    * @param newProjectName
+    *       new name of the project
+    */
    public void renameProject(final String projectId, final String newProjectName) {
-
+      DataDocument dataDocument = new DataDocument(Project.ATTR_PROJECT_NAME, newProjectName);
+      dataStorage.updateDocument(Project.COLLECTION_NAME, dataDocument, projectIdFilter(projectId));
    }
 
+   /**
+    * Creates new project
+    *
+    * @param projectId
+    *       id of the project to create
+    * @param projectName
+    *       name of the project
+    */
    public void createProject(final String projectId, final String projectName) {
+      DataDocument document = new DataDocument(Project.ATTR_PROJECT_ID, projectId)
+            .append(Project.ATTR_PROJECT_NAME, projectName)
+            .append(Project.ATTR_ORGANIZATION_ID, organisationFacade.getOrganisationId());
+      dataStorage.createDocument(Project.COLLECTION_NAME, document);
+   }
 
+   /**
+    * Drops the project if it hasn't assigned any collection or view
+    *
+    * @param projectId
+    *       id of the project to drop
+    */
+   public void dropProject(final String projectId) {
+      dataStorage.dropDocument(Project.COLLECTION_NAME, projectIdFilter(projectId));
+      // TODO check cannot be implemented without other facades changes
    }
 
-   public void deleteProject(final String projectId) {
+   /**
+    * Reads default project roles for newly added users
+    *
+    * @param projectId
+    *       id of the project
+    * @return List of default user roles
+    */
+   public List<String> readDefaultRoles(final String projectId) {
+      DataDocument document = dataStorage.readDocumentIncludeAttrs(Project.COLLECTION_NAME, projectIdFilter(projectId), Collections.singletonList(Project.ATTR_META_DEFAULT_ROLES));
+      return document != null ? document.getArrayList(Project.ATTR_META_DEFAULT_ROLES, String.class) : null;
+   }
 
+   /**
+    * Set default roles for newly added users
+    *
+    * @param projectId
+    *       id of the project
+    * @param userRoles
+    *       List of the user roles to set
+    */
+   public void setDefaultRoles(final String projectId, final List<String> userRoles) {
+      DataDocument document = new DataDocument(Project.ATTR_META_DEFAULT_ROLES, userRoles);
+      dataStorage.updateDocument(Project.COLLECTION_NAME, document, projectIdFilter(projectId));
    }
 
-   public void setDefaultRolesToProject(final String projectId, final List<String> userRoleIds) {
+   /**
+    * Create new user role which consists of core roles
+    *
+    * @param projectId
+    *       Id of the project
+    * @param userRole
+    *       Unique name of user role
+    * @param coreRoles
+    *       List of the core roles to set
+    */
+   public void createRole(final String projectId, final String userRole, final List<String> coreRoles) {
```

Comment:
I think all role management methods should rather be in `UserFacade` or something similar and not in `ProjectFacade`.